src/prng/isaac64.rs

```diff
@@ -181,36 +188,82 @@ impl Isaac64Rng {
 
         self.a = a;
         self.b = b;
-        self.cnt = RAND_SIZE as u32;
+        self.cnt = (RAND_SIZE * 2) as u32;
+    }
+
+    fn fill_chunk(&mut self, dest: &mut [u8]) -> usize {
+        if self.cnt < 2 {
+            self.isaac64();
+        }
+
+        let mut index_u64 = (self.cnt >> 1) as usize;
+        let available = index_u64 * 8;
```

Comment:
Why not just use `index_u32` here and keep any extra half-word?

I tried implementing this, and ran into a complication: the Endian conversion still works on 64-bit words. Dealing with this the hard way (rounding up/down and making sure to discard any extra converted half-word) is possible, but adds complication. Benchmark shows no difference... well, it shouldn't really, especially if I don't try throwing away half-words.

So better just forget this idea!

Reply:
Before I had the endian conversion that was how it worked. With the conversion it just got a bit difficult. It only matters if you frequently call `fill_bytes` after `next_u32`, and thought it was not worth the extra complexity.